python/core/auto_generated/raster/qgsrasterlayer.sip.in

```diff
@@ -258,6 +258,20 @@ Returns the name of a band given its number.
 %Docstring
 Returns the (possibly NULL) raster attribute table for the given band ``bandNumber``.
 
+.. versionadded:: 3.30
+%End
+
+    int attributeTableCount( ) const;
+%Docstring
+Returns the number of attribute tables for the raster by counting the number of bands that have an associated attribute table.
+
+.. versionadded:: 3.30
+%End
+
+    bool canCreateRasterAttributeTable( );
```

Comment:
I'd rather leave it there, we might decide to add further checks like the raster being writable in case of embedded RATs.
Currently, the capability to create a RAT depends on the renderer type but I'd say this is an implementation detail and additional checks might be added in the future.